app/src/test/java/ingsw/group1/findmyphone/event/SMSLogDatabaseTest.java

```diff
@@ -0,0 +1,220 @@
+            LogEventType.UNKNOWN,
+            EXAMPLE_CONTACT,
+            100L,
+            "Hello"
+    );
+    private static final SMSLogEvent ANOTHER_EVENT = new SMSLogEvent(
+            LogEventType.RING_REQUEST_RECEIVED,
+            EXAMPLE_CONTACT,
+            10000L,
+            String.valueOf(1000L)
+    );
+    private static final List<SMSLogEvent> SOME_EVENTS = Arrays.asList(
+            SIMPLE_EVENT, ANOTHER_EVENT
+    );
+
+    private SMSLogDatabase database;
+
+    /**
+     * Loading an instance of the database.
+     */
+    @Before
+    public void loadDatabase() {
+        database = SMSLogDatabase.getInstance(
+                ApplicationProvider.getApplicationContext(),
+                DEFAULT_DB_NAME
+        );
+    }
+
+    /**
+     * Just a facade call to allow actual clearing method to be static, to be used in any other
+     * tests that need it.
+     */
+    @After
+    public void clearInstances() {
+        clearActiveInstances();
+    }
+
+    /**
+     * This method is necessary, due to an issue with ObjectPools where static instances of the
+     * database cannot be reset between tests automatically.
+     */
+    public static void clearActiveInstances() {
+        try {
+            //Getting the field named "activeInstances" (the map with all active databases)
+            Field instance = SMSLogDatabase.class.getDeclaredField("activeInstances");
+            //Making it accessible
+            instance.setAccessible(true);
+            //Casting it to a Map
+            Map<?, ?> privateField = (Map<?, ?>) instance.get(null);
+            //Making it empty
+            if (privateField != null)
+                privateField.clear();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Test confirming same name results in same instance.
+     * Testing for the hashcode is implicit because the instances are the exact same.
+     */
+    @Test
+    public void sameNameAreSame() {
+        assertSame(
+                SMSLogDatabase.getInstance(
+                        ApplicationProvider.getApplicationContext(),
+                        DEFAULT_DB_NAME
+                ),
+                SMSLogDatabase.getInstance(
+                        ApplicationProvider.getApplicationContext(),
+                        DEFAULT_DB_NAME
+                )
+        );
+    }
+
+    /**
+     * Test confirming different name results in different instance.
+     */
+    @Test
+    public void differentNameAreNotEqual() {
+        SMSLogDatabase anInstance = SMSLogDatabase.getInstance(
+                ApplicationProvider.getApplicationContext(),
+                DEFAULT_DB_NAME
+        );
+        SMSLogDatabase anotherInstance = SMSLogDatabase.getInstance(
+                ApplicationProvider.getApplicationContext(),
+                ALTERNATIVE_DB_NAME
+        );
+        assertTrue(
+                anInstance != anotherInstance &&
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4969671)